consensus/block_validation.go

```diff
@@ -0,0 +1,87 @@
+package consensus
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+// BlockValidator defines an interface used to validate a blocks syntax and
+// semantics.
+type BlockValidator interface {
+	BlockSemanticValidator
+	BlockSyntaxValidator
+}
+
+// BlockSemanticValidator defines an interface used to validate a blocks
+// semantics.
+type BlockSemanticValidator interface {
+	ValidateSemantic(ctx context.Context, child, parent *types.Block) error
+}
+
+// BlockSyntaxValidator defines an interface used to validate a blocks
+// syntax.
+type BlockSyntaxValidator interface {
+	ValidateSyntax(ctx context.Context, blk *types.Block) error
+}
+
+// BlockValidationClock defines an interface for fetching unix epoch time.
+type BlockValidationClock interface {
+	EpochSeconds() uint64
+}
+
+// DefaultBlockValidationClock implements BlockValidationClock using the
+// Go time package.
+type DefaultBlockValidationClock struct{}
+
+// NewDefaultBlockValidationClock returns a DefaultBlockValidationClock.
+func NewDefaultBlockValidationClock() *DefaultBlockValidationClock {
+	return &DefaultBlockValidationClock{}
+}
+
+// EpochSeconds returns Unix time, the number of seconds elapsed since January 1, 1970 UTC.
+// The result does not depend on location.
+func (ebc *DefaultBlockValidationClock) EpochSeconds() uint64 {
+	return uint64(time.Now().Unix())
+}
+
+// DefaultBlockValidator implements the BlockValidator interface.
+type DefaultBlockValidator struct {
+	clock     BlockValidationClock
+	blockTime time.Duration
+}
+
+// NewDefaultBlockValidator returns a new DefaultBlockValidator. It uses `blkTime`
+// to validate blocks and uses the DefaultBlockValidationClock.
+func NewDefaultBlockValidator(blkTime time.Duration) *DefaultBlockValidator {
+	return &DefaultBlockValidator{
+		clock:     NewDefaultBlockValidationClock(),
+		blockTime: blkTime,
+	}
+}
+
+// ValidateSemantic validates a block is correctly derived from its parent.
+func (dv *DefaultBlockValidator) ValidateSemantic(ctx context.Context, child, parent *types.Block) error {
```

Comment:
Blocking: `parent` cannot be a block.  When we fill this in with all of the needed validation we are going to need to operate on the whole parent which is a tipset.  For example we will need the min ticket.  As a more relevant example, we will need the min timestamp of the parent tipset to do the soft block time validation.